### PR TITLE
update token validation to remove VQR

### PR DESCRIPTION
**Pull request by sfc-gh-jhilgart**
Exclude VQR from total token count limit.

### CODE REVIEW

File: semantic_model_generator/validate/context_length.py [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+
+    yaml_str: The yaml semantic model
+    throw_error: Should this function throw an error or just a warning.
+    remove_vqr: Whether to remove the VQR from total token count. As these are indexed separately, default to True.
+    """
+    # Convert back to a model to remove sections that don't count against token limit (i.e. VQR).
+    yaml_str_copy = copy.deepcopy(yaml_str)
+    if remove_vqr:
+        model = yaml_to_semantic_model(yaml_str_copy)
+        model.ClearField("verified_queries")
```
Comment (sfc-gh-rehuang) [inline semantic_model_generator/validate/context_length.py:28]: add test on whether removed VQR successfully?

File: semantic_model_generator/validate/context_length.py [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
+    yaml_to_semantic_model,
+)
+
+_MODEL_CONTEXT_LENGTH_TOKENS = 6500  # We use 6.5k, with 1.2k for instructions, so that we can reserve 500 for response tokens (average is 300).
 _MODEL_CONTEXT_INSTR_TOKEN = 20  # buffer for instr tokens
 
 
-def validate_context_length(yaml_str: str, throw_error: bool = True) -> None:
+def validate_context_length(
+    yaml_str: str, throw_error: bool = True, remove_vqr: bool = True
```
Comment (sfc-gh-rehuang) [inline semantic_model_generator/validate/context_length.py:15]: put throw_error default as False?

Comment (sfc-gh-rehuang) [reply semantic_model_generator/validate/context_length.py:15]: and `remove_vqr` probably don't need to be a param... can't think of cases that we don't want to remove vqr

File: semantic_model_generator/validate/context_length.py (hunk 1 above)
Comment (sfc-gh-rehuang) [reply semantic_model_generator/validate/context_length.py:28]: this conversion here seems unnecessary. Just change the input to take in a proto model?
places where this function's called has conversion to models anyways
https://github.com/Snowflake-Labs/semantic-model-generator/blob/c45485e46fb0bb46e15f77b4b62b3ecefecafae8/semantic_model_generator/validate_model.py#L36

https://github.com/Snowflake-Labs/semantic-model-generator/blob/c45485e46fb0bb46e15f77b4b62b3ecefecafae8/semantic_model_generator/generate_model.py#L305

Comment (sfc-gh-jhilgart) [reply semantic_model_generator/validate/context_length.py]: The test `test_valid_yaml_with_long_vqr_context -` fails if we don't clear VQR

File: semantic_model_generator/validate/context_length.py (hunk 2 above)
Comment (sfc-gh-jhilgart) [reply semantic_model_generator/validate/context_length.py]: done

**Review by sfc-gh-rehuang — APPROVED** [review]
thank you!

**Pull request merged**